### PR TITLE
Fix login documentation layout

### LINKED ISSUE #1413: `login` documentation strange headlines
**Description**

The `login` markdown documentation has strange headlines.

https://docs.docker.com/engine/reference/commandline/login/

ddadd3db494ce568a6b847932677f837802ea6b5 mass standardized the
formatting, with some errors, notably in `login.md`:
- `Logging out` headline is wrong (it's just that `docker logout` is a prerequisite to the `Usage` documented in the following lines):  https://github.com/docker/cli/commit/ddadd3db494ce568a6b847932677f837802ea6b5#diff-c033ce901f31496d94f0e354c81a9cdbR139 
- some headlines were shifted one level but not their own nested headlines
- nested `Usage` headlines were removed, making the surrounding text less comprehensible 

I have a PR coming for this issue.

I am not sure if we want to restore the nested `Usage` headlines, should we?

### PR DESCRIPTION
**Pull request by thomas-riccardi**
closes #1413 

**- What I did**
ddadd3db494ce568a6b847932677f837802ea6b5 mass standardized the
formatting, with some errors.

This commit fixes errors on `login.md`

- revert wrong `Logging out` headline
- restore correct level for some headlines (relative to parent
  headline level change)
- re-add Usage

**- How I did it**
.

**- How to verify it**
Check rendered documentation

**- Description for the changelog**
Better `login` documentation layout

### CODE REVIEW

File: docs/reference/commandline/login.md [hunk 1]
```diff
@@ -85,6 +85,8 @@ you can download them from:
 - Microsoft Windows Credential Manager: https://github.com/docker/docker-credential-helpers/releases
 - [pass](https://www.passwordstore.org/): https://github.com/docker/docker-credential-helpers/releases
 
+#### Usage
```
Comment (thaJeztah) [inline docs/reference/commandline/login.md]: Thinking if we should use a more descriptive title; perhaps "Credentials store configuration", or "Configure the credentials store" ?

(open to suggestions 😅)

File: docs/reference/commandline/login.md [hunk 2]
```diff
@@ -162,7 +164,7 @@ designated programs to handle credentials for *specific registries*. The default
 credential store (`credsStore` or the config file itself) will not be used for
 operations concerning credentials of the specified registries.
 
-### Logging out
+#### Usage
```
Comment (thaJeztah) [inline docs/reference/commandline/login.md]: Thinking if we should use a more descriptive title; perhaps "Credential helper configuration", or "Configure credential helpers" ?

File: docs/reference/commandline/login.md [hunk 3]
```diff
@@ -162,7 +164,7 @@ designated programs to handle credentials for *specific registries*. The default
 credential store (`credsStore` or the config file itself) will not be used for
 operations concerning credentials of the specified registries.
 
-### Logging out
```
Comment (thaJeztah) [inline docs/reference/commandline/login.md]: Good catch; would be worth to add a "Related commands" at the end of this file, to link to `logout` (similar to https://github.com/docker/cli/blob/master/docs/reference/commandline/plugin_create.md#related-commands)

File: docs/reference/commandline/login.md [hunk 4]
```diff
@@ -162,7 +164,7 @@ designated programs to handle credentials for *specific registries*. The default
...
+#### Usage
 
 If you are currently logged in, run `docker logout` to remove
 the credentials from the default store.
```
Comment (thaJeztah) [inline docs/reference/commandline/login.md]: If you're interested in making some additional changes (could be in a follow-up); I think we should include a link somewhere to the section describing where the cli configuration file is; https://github.com/docker/cli/blob/master/docs/reference/commandline/cli.md#configuration-files)

File: docs/reference/commandline/login.md (hunk 1 above)
Comment (thomas-riccardi) [reply docs/reference/commandline/login.md]: I went for the `Configure the ...` way. Both ways already existed:
```
run.md:787:### Configure namespaced kernel parameters (sysctls) at runtime
pull.md:40:### Proxy configuration
```

File: docs/reference/commandline/login.md (hunk 2 above)
Comment (thomas-riccardi) [reply docs/reference/commandline/login.md]: done

File: docs/reference/commandline/login.md (hunk 3 above)
Comment (thomas-riccardi) [reply docs/reference/commandline/login.md]: done

File: docs/reference/commandline/login.md (hunk 4 above)
Comment (thomas-riccardi) [reply docs/reference/commandline/login.md]: I'll probably do that in a follow-up PR soon.

**Pull request merged**